Cli/SendReports.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Authors: Alex Gusev <alex@flancer64.com>
+ * Since: 2020
+ */
+
+namespace Flancer32\Csp\Cli;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Analyze CSP violation reports and compose policy rules.
+ */
+class SendReports
+    extends \Flancer32\Base\App\Cli\Base
+{
+    private const DESC = 'reports new rules to specified mail addresses and registers reported rules persistently in protocol table';
+    private const NAME = 'fl32:csp:report_new_rules';
+
+    /** @var \Flancer32\Csp\Helper\Config */
+    private $hlpCfg;
+    /**
+     * @var \Flancer32\Csp\Model\Service\ReportNewRules
+     */
+    private $reportNewRules;
+
+    public function __construct(
+        \Flancer32\Csp\Helper\Config $hlpCfg,
+        \Flancer32\Csp\Model\Service\ReportNewRules $reportNewRules
+    ) {
+        parent::__construct(self::NAME, self::DESC);
+        $this->reportNewRules = $reportNewRules;
+        $this->hlpCfg = $hlpCfg;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        /* parse and validate input data */
+
+        $output->writeln("<info>Command '" . $this->getName() . "': " . $this->getDescription() . "<info>");
+        $this->checkAreaCode();
+        if ($this->hlpCfg->getEnabled()) {
+            try {
+                $this->reportNewRules->execute($throwException = true);
+            } catch (\Exception $exception) {
+                $output->writeln('<info>' . $exception->getMessage() . '<info>');
+            }
+
+        } else {
```

Comment:
a general advice: it's considered good practice to put checks like this in the beginning and exit/return early, instead of down in an `else` branch. E.g.
```
if (! enabled) {
  echo "nope"
  return;
}
// actual code here
```

**Advantages:**
- guard clauses at the beginning are visually separated from actual execution
- less indentation, less complexity

Reply:
new terms learned, thank you